src/resources/migrating-to-material-3.md

```diff
@@ -0,0 +1,513 @@
+---
+title: Migrating to Material 3
+description: >-
+  Migration guide for updating your UI from Material 2 to Material 3.
+---
+
+## Summary
+
+The Material library has been updated to match the Material 3 Design
+spec.  Changes include new components and component themes,
+updated component visuals, and much more. Many of these updates
+are seamless. You’ll see the new version of the widget when recompiling
+your app against the 3.16 (or later) release. But some manual work is also
+required to complete the migration.
+
+## Migration guide
+
+Prior to the 3.16 release, you could opt in to the Material 3 changes by
+setting `useMaterial3` flag to true. As of the Flutter 3.16 release
+(November 2023), useMaterial3 is true by default.
+
+By the way, you _can_ recapture Material 2 behavior in your app by setting
+`useMaterial3` to `false`. However, this is just a temporary solution. The
+`useMaterial3` flag _and_ the Material 2 implementation will eventually be
+removed as part of Flutter’s deprecation policy.
```

Comment:
@HansMuller 
This was written before your [comment](https://github.com/flutter/flutter/issues/141555#issuecomment-1896296465). Is this still accurate?

 If you have any suggestions on updating this.

Reply:
It is still true. Before we deprecate M2 we'll provide a theme, in a separate public package, that closely approximates the M2 defaults.